lhotari/pulsar
# lh quarantine tests experiment

- Move most flaky tests to "quarantine" test group
- Use "alwaysRun=true" for some BeforeMethods
- Use alwaysRun=true for all BeforeMethod annotations
- Add retries for BkEmsemblesTestBase
- Add retry support for ClientDeduplicationFailureTest
- Fix slow initialization
- Add testDeadLetterTopic to quarantine
- Remove -DtestReuseFork=false from running flaky tests
- Disable verbose output when running quarantined tests
- Use testForkCount=2 for all unit tests
- Use testForkCount=1 for shade and backwards compat tests
- Don't reuse forks for integration tests
- Remove retry_integration.sh since there is no need to kill Docker containers
- Fix test initialization
- Use -DtestForkCount=2 for all mvn command in run_unit_group.sh script
- Add null check
- Use consistent method name (same as the other similar case)
- Print TestNG failures to console output and as GitHub Action warnings
- Fail fast when running tests
- Increase timeout for broker group 1
- Enable test retries for integration tests, shade tests and backwards compat tests
